lib/structuraid_core/elements/footing.rb

```diff
@@ -104,12 +96,53 @@ def inside_me?(location)
         inside_axis_1 && inside_axis_2
       end
 
+      # Returns the length of the footing parallel to the given section
+      # @param section_direction [Symbol] The direction of the working section
+      # @return [Float] The length of the footing in the perpendicular direction to the given section
       def width(section_direction:)
+        return @length_1 if section_direction == :length_1
+
+        @length_2
+      end
+
+      # Returns the length of the footing in the perpendicular direction to the given section
+      # @param section_direction [Symbol] The direction of the working section
+      # @return [Float] The length of the footing in the perpendicular direction to the given section
+      def lenght(section_direction:)
```

Comment:
There is a typo on the method name. Also, for these simple methods, I think keyword arguments are more a nuisance rather than a help. It makes things longer and harder to read when we call them and I don't think they help making the code more clear. What do you think?

```suggestion
      def length(section_direction)
```

Reply:
Done!